### DISCUSSION TITLE
Tree Shaking mechanism for SEA

### DISCUSSION THREAD

**Discussion by bayraak** [nodejs/single-executable#62] [General]
The simple `hello world` program made with Node SEA is about `~90 MB` on macOS. Is there any plan or feature to have a tree-shaking mechanism that cuts the not used Node API's or system functions?

**Comment by RaisinTen** [comment]
Currently, I don't think we don't have a feature for trimming out the unused parts from Node.js but if you build Node.js with some configuration options like `--without-intl`, `--without-ssl`, etc. based on your needs, you might be able to create a smaller SEA.

**Reply by Link2Twenty** [reply]
Is there a list of all the `--without` flags in the documentation?

**Reply by RaisinTen** [reply]
@Link2Twenty you could try taking a look at the output of this command in the Node.js repo - `./configure --help`. It prints all the configuration options of Node.js with information about what those options mean. In case you're wondering, this is where these options are defined - https://github.com/nodejs/node/blob/472ffdef73f636cefa0e7015e07499637c34bd43/configure.py#L81-L843.
